notif.go

```diff
@@ -93,21 +93,6 @@ func (nn *netNotifiee) Disconnected(n inet.Network, v inet.Conn) {
 	}
 
 	dht.routingTable.Remove(p)
-
-	dht.smlk.Lock()
```

Comment:
Where are we handling disconnects now? It feels like `streamPool` will retain disconnected peers forever?

Reply:
Ah, false alarm. We're relying on the stream closing: https://github.com/libp2p/go-libp2p-kad-dht/pull/285/files#diff-e2d91404d5609adc9273e4603f6ce67dR27 to delete the stream from the pool. And yeah, I guess we can safely assume that when a peer disconnects, the swarm closes all its streams.

Reply:
That's right.